### PR TITLE
Clean up clearing faults and add verbose output on failure

### PR DESCRIPTION
**Pull request by MarqRazz**
This PR adds delays to the fault clearing process to ensure the arm has time to process each command. It also adds verbose output when clearing the fault is unsuccessful.

@johnsonTyler0511 when you get a chance could you please test this out on hardware and let me know how it works?

### CODE REVIEW

**Comment by johnsonTyler0511** [comment]
Looks like some good changes, ill test on hardware soon :1st_place_medal:

File: kortex_driver/src/hardware_interface.cpp [hunk 1]
```diff
@@ -1040,6 +1006,125 @@ void KortexMultiInterfaceHardware::sendTwistCommand()
...
 
+void KortexMultiInterfaceHardware::resetFaults()
+{
+  if (in_fault_ == 0.0)
+  {
+    reset_fault_async_success_ = 1.0;
+    reset_fault_cmd_ = NO_CMD;  // clear the reset_fault_cmd
+    RCLCPP_INFO(LOGGER, "Arm is not in a faulted state. Returning SUCCESS from resetFaults().");
+    return;
+  }
```
Comment (forrest-rm) [inline kortex_driver/src/hardware_interface.cpp:1018]: Do we expected to be in this state? The previous code didn't check for this, and restultFaults only gets called if `!std::isnan(reset_fault_cmd_)`.

Depending on how often we get here, the log output may be spammed.

File: kortex_driver/src/hardware_interface.cpp [hunk 2]
```diff
@@ -1040,6 +1006,125 @@ void KortexMultiInterfaceHardware::sendTwistCommand()
...
+  try
+  {
+    // change servoing mode first
+    servoing_mode_hw_.set_servoing_mode(k_api::Base::ServoingMode::SINGLE_LEVEL_SERVOING);
+    base_.SetServoingMode(servoing_mode_hw_);
+    // wait for the switch to apply
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
```
Comment (forrest-rm) [inline kortex_driver/src/hardware_interface.cpp]: Please create a constant variable defined at the top of the file for the reset wait time.

File: kortex_driver/src/hardware_interface.cpp [hunk 3]
```diff
@@ -1040,6 +1006,125 @@ void KortexMultiInterfaceHardware::sendTwistCommand()
...
+      k_api::ErrorCodes_Name(ex.getErrorInfo().getError().error_code()).c_str());
+    RCLCPP_ERROR(
+      LOGGER, "Error sub code: %s\n",
+      k_api::SubErrorCodes_Name(k_api::SubErrorCodes(ex.getErrorInfo().getError().error_sub_code()))
+        .c_str());
+    RCLCPP_ERROR(LOGGER, "Error description: %s\n", ex.what());
+    reset_fault_async_success_ = 0.0;
+    reset_fault_cmd_ = NO_CMD;  // clear the reset_fault_cmd
+  }
+  catch (std::runtime_error & ex_runtime)
```
Comment (forrest-rm) [inline kortex_driver/src/hardware_interface.cpp:1119]: The previous version had a default catch statement. Is that no longer needed?

**Review by forrest-rm — COMMENTED** [review]
Code changes look good pending hardware tests. At some point, we discuss which changes we may want to push to the upstream branch.

File: kortex_driver/src/hardware_interface.cpp (hunk 3 above)
Comment (johnsonTyler0511) [reply kortex_driver/src/hardware_interface.cpp:1119]: I could be wrong but I think this was there to catch the other possible exceptions other than `KDetailedException` and `runtime_error`, I think the `base_cyclic_.RefreshFeedback()` might also throw a `std::future_error` and a `std::exception` based on the other try catch blocks in the code.

**Comment by johnsonTyler0511** [comment]
Testing with Dane on bench4 waiting for a fault :+1:

File: kortex_driver/src/hardware_interface.cpp (hunk 3 above)
Comment (bmagyar) [reply kortex_driver/src/hardware_interface.cpp:1119]: Catching std::exception is always a good idea as it gets you a `what()` to print. Doing `catch(...)` is just sending it down the drain without any signal of what's up, not really good for the long run

**Comment by johnsonTyler0511** [comment]
Tested this on bench4 was able to reset from 4 faults when I was stressing the system, needed to restart pick client tho

**Comment by MarqRazz** [comment]
@johnsonTyler0511  Can we force a collision fault and see if this is able to recover?

**Review by johnsonTyler0511 — APPROVED** [review]
Tested on hardware and reset is much more robust, was able to reset from the faults consistently when missing rate. Looks good to me pending other comments

**Comment by johnsonTyler0511** [comment]
> @johnsonTyler0511 Can we force a collision fault and see if this is able to recover?

Okay I can try to do that

**Comment by johnsonTyler0511** [comment]
> > @johnsonTyler0511 Can we force a collision fault and see if this is able to recover?
> 
> Okay I can try to do that

yep looks good even when I physically stop it :1st_place_medal:

**Comment by johnsonTyler0511** [comment]
It def is much more robust and have not seen it clear as many faults before but I did just get a fault that it did not clear successfully and still got into a stuck state where I needed to restart the kinova arm. Not sure if this was the original fault as the arm could've moved a bit after resetting but it was joint 3 position limit high

File: kortex_driver/src/hardware_interface.cpp (hunk 1 above)
Comment (MarqRazz) [reply kortex_driver/src/hardware_interface.cpp:1018]: If the service is called when there is not a fault I didn't want the arm to be e-stopped so I added this guard.

File: kortex_driver/src/hardware_interface.cpp (hunk 3 above)
Comment (MarqRazz) [reply kortex_driver/src/hardware_interface.cpp:1119]: I updated the code to include the blanket catch with a print that asks the developer to update the code to catch it properly.

**Pull request merged**